xm-commons-config/src/main/java/com/icthh/xm/commons/config/client/service/ConfigCacheFactory.java

```diff
@@ -0,0 +1,71 @@
+package com.icthh.xm.commons.config.client.service;
+
+import com.icthh.xm.commons.config.client.api.RefreshableConfiguration;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+
+import static java.util.Collections.unmodifiableMap;
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class ConfigCacheFactory implements RefreshableConfiguration {
```

Comment:
@sergeysenja1992 it worth thinking about making this caching service 'tenant aware'. What I mean now each tenant can get the value from another tenant just by guessing its key name. If we will add one more level with the tenant to the refreshableConfigurations we can handle tenant selection by TenantContextHolder.